metrics/micrometer/src/main/java/com/datastax/oss/driver/internal/metrics/micrometer/MicrometerNodeMetricUpdater.java

```diff
@@ -111,8 +111,17 @@ protected Timer.Builder configureTimer(Timer.Builder builder, NodeMetric metric,
                   : null)
           .percentilePrecision(
               profile.getInt(DefaultDriverOption.METRICS_NODE_CQL_MESSAGES_DIGITS));
+
+      if (profile.isDefined(DefaultDriverOption.METRICS_NODE_CQL_MESSAGES_PUBLISH_PERCENTILES)) {
+        builder.publishPercentiles(
+            profile.getDoubleList(DefaultDriverOption.METRICS_NODE_CQL_MESSAGES_PUBLISH_PERCENTILES)
+                .stream()
+                .mapToDouble(Double::doubleValue)
+                .toArray());
+      }
```

Comment:
There's already a lot of repetition in this file but let's start to minimize it by abstracting this out to a helper method:

```java
private double[] toDoubleArray(List<Double> doubleList) {                                          
  return doubleList.stream().mapToDouble(Double::doubleValue).toArray();               
} 
```

I would've thought we could have avoided some of the intermediate object creation here by just calling publishPercentiles() multiple times on the builder:

```java
       if (profile.isDefined(DseDriverOption.METRICS_SESSION_GRAPH_REQUESTS_PUBLISH_PERCENTILES)) {
         profile          
           .getDoubleList(DseDriverOption.METRICS_SESSION_GRAPH_REQUESTS_PUBLISH_PERCENTILES)
           .forEach(builder::publishPercentiles);
       } 
```

But weirdly that didn't work.  With this change the unit tests fail to pass, claiming that this assert:

```java
    assertThat(snapshot.percentileValues()).hasSize(3);
```

is only finding one item rather than three... which suggests that percentiles aren't just added together.

Reply:
added helper method. let me know anything else needed

Reply:
created separate static method